File: HackerRank-10-Days-Statistics/Day 5/Day_5_Normal_Distribution_I.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include<iomanip>
#include<bits/stdc++.h>
using namespace std;

double cum_dist(double x,double mean, double dev){
    double dist =0.5*(1+ erf((x-mean)/(sqrt(2)*dev))) ;
    return dist;
}


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    double mean, dev, val, lower_bound, upper_bound;
    cin>>mean>>dev>>val>>lower_bound>>upper_bound;
    
    cout<<fixed<<setprecision(3)<<cum_dist(val,mean,dev)<<endl;
    cout<<fixed<<setprecision(3)<<(cum_dist(upper_bound,mean,dev)-cum_dist(lower_bound,mean,dev))<<endl;
    //ed)<<normal_dist(upper_bound,mean,dev)-normal_dist(lower_bound,mean,dev)<<endl;
    
       
    return 0;
}

File: HackerRank-10-Days-Statistics/Day 5/Day_5_Normal_Distribution_I.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import math
import sys

a,b,c = sys.stdin.readlines()
mean, dev = list(map(float,a.strip().split()))
val=float(b)
lowr_bound,uppr_bound=list(map(float,c.strip().split()))

def cum_dist(x):
    dist=0.5*(1+ math.erf((x-mean)/(dev*math.sqrt(2))))
    return dist

print(round(cum_dist(val),3))
print(round(cum_dist(uppr_bound)-cum_dist(lowr_bound),3))